olm/catalogd/v1alpha1/clustercatalog_types.go

```diff
@@ -0,0 +1,223 @@
+	//
+	// Below is a minimal example of a ClusterCatalogSpec that sources a catalog from an image:
+	//
+	//  source:
+	//    type: Image
+	//    image:
+	//      ref: quay.io/operatorhubio/catalog:latest
+	//
+	// For more information on FBC, see https://olm.operatorframework.io/docs/reference/file-based-catalogs/#docs
+	Source CatalogSource `json:"source"`
+
+	// priority is an optional field that allows the user to define a priority for a ClusterCatalog.
+	// A ClusterCatalog's priority is used by clients as a tie-breaker between ClusterCatalogs that meet the client's requirements.
+	// For example, in the case where multiple ClusterCatalogs provide the same bundle.
+	// A higher number means higher priority. Negative numbers are also accepted.
+	// When omitted, the default priority is 0.
+	// +kubebuilder:default:=0
+	// +optional
+	Priority int32 `json:"priority"`
+
+	// Availability is an optional field that allows users to define whether the ClusterCatalog is utilized by the operator-controller.
+	//
+	// Allowed values are : ["Enabled", "Disabled"].
+	// If set to "Enabled", the catalog will be used for updates, serving contents, and package installations.
+	//
+	// If set to "Disabled", catalogd will stop serving the catalog and the cached data will be removed.
+	//
+	// If unspecified, the default value is "Enabled"
+	//
+	// +kubebuilder:validation:Enum="Disabled";"Enabled"
+	// +kubebuilder:default="Enabled"
+	// +optional
+	Availability string `json:"availability,omitempty"`
+}
+
+// ClusterCatalogStatus defines the observed state of ClusterCatalog
+type ClusterCatalogStatus struct {
+	// conditions is a representation of the current state for this ClusterCatalog.
+	// The status is represented by a set of "conditions".
+	//
+	// Each condition is generally structured in the following format:
+	//   - Type: a string representation of the condition type. More or less the condition "name".
+	//   - Status: a string representation of the state of the condition. Can be one of ["True", "False", "Unknown"].
+	//   - Reason: a string representation of the reason for the current state of the condition. Typically useful for building automation around particular Type+Reason combinations.
+	//   - Message: a human-readable message that further elaborates on the state of the condition.
+	//
+	// The current set of condition types are:
+	//   - "Serving", which represents whether or not the contents of the catalog are being served via the HTTP(S) web server.
+	//   - "Progressing", which represents whether or not the ClusterCatalog is progressing towards a new state.
+	//
+	// The current set of reasons are:
+	//   - "Succeeded", this reason is set on the "Progressing" condition when progressing to a new state is successful.
+	//   - "Blocked", this reason is set on the "Progressing" condition when the ClusterCatalog controller has encountered an error that requires manual intervention for recovery.
```

Comment:
You are correct that these both result in `Progressing` being set to `False`. If I am remembering correctly, we only have the `Blocked` reason to handle the case where an invalid image pull spec is provided meaning direct user intervention is required to fix the issue. 

I think adding constraints to the `ClusterCatalog.Spec.Source.Image.Ref` field would allow us to be more confident that the pull spec is a valid format and alleviate a "terminal" error state like this.

@joelanford please correct me if my understanding of why we added this `Blocked` reason is wrong.

Reply:
Does that sound reasonable or are there other examples that you think might be better to take a look at?

Reply:
One thing to consider is that progressing in openshift operators is traditionally false when at a level, but also has a sad state that is false.

The key difference I see between that and your design is that it has other conditions (like available) that would signal to the user that something is wrong.

Perhaps adding an Available condition here would help? Would it mean that you could have `Available: False, Progressing: False` as a sad state, but `Available: True, Progressing: False` as the steady state and `Available: True, Progressing: True` as ???

What does it actually mean for progressing to be true here?

Reply:
> Our Available is called Serving. If we had both Available and Serving that would be pretty confusing, I think.

Agreed, I saw that on another thread after replying to this one

So in a steady state, at a glance I can see `Available: True`, and `Progressing: False`, now, how do I represent a failure in progressing? Today it looks the same, is adding something that indicates if we are the desired version the way to represent the difference?

It feels today like Serving is useful to know that something is working, but, doesn't give me an indication of whether we have reached the desired state

Reply:
Is using a `Reason` that reflects that there is a failure/success in progressing insufficient to indicate this?
In your experience do users default to looking only at condition statuses of True/False to signal that difference?

Reply:
> It feels today like Serving is useful to know that something is working, but, doesn't give me an indication of whether we have reached the desired state

Serving: true is the desired state. We want catalogs to be in serving state for users to consume them. may be we should be explicit about the desired state in the go-doc.

Reply:
> Serving: true is the desired state

Not necessarily. We could be successfully serving an old catalog while simultaneously failing to pull the catalog stated in the spec.

Reply:
We settled on updating the `Progressing` condition such that:
- Reasons `Succeeded` and `Retrying` are considered happy and possible when status is True
- Reason `Blocked` is considered sad and set when status is False
- In the near future (v1.1), add a retry limit feature that when the retry limit is exceeded, this condition is set to False with a reason that reflects a retry limit failure.